ramalama/ollama.py

```diff
@@ -87,6 +86,16 @@ def pull_blob(repos, layer_digest, accept, registry_head, models, model_name, mo
     headers = {"Accept": accept}
     download_file(url, layer_blob_path, headers=headers)
 
+    # Verify checksum after downloading the blob
+    if not verify_checksum(layer_blob_path):
+        print(f"Checksum mismatch for blob {layer_blob_path}, retrying download...")
+        os.remove(layer_blob_path)
+        download_file(url, layer_blob_path, headers=headers)
+        if not verify_checksum(layer_blob_path):
+            raise ValueError(f"Checksum verification failed for blob {layer_blob_path}")
+    else:
+        print(f"Checksum verified for blob {layer_blob_path}.")
```

Comment:
This is going to be noisy.  I think humans will just assume what they pulled was correct, and this message will not add anything.